actioncable/lib/action_cable/server/base.rb

```diff
@@ -81,9 +108,14 @@ def worker_pool
         @worker_pool || @mutex.synchronize { @worker_pool ||= ActionCable::Server::Worker.new(max_size: config.worker_pool_size) }
       end
 
+      # Executor is used by various actions within Action Cable (e.g., pub/sub operations) to run code asynchronously.
+      def executor
+        @executor || @mutex.synchronize { @executor ||= ThreadedExecutor.new(max_size: config.executor_pool_size) }
```

Comment:
Maybe this - and the pool size, should be provided entirely by the user, e.g. a traits class.

Reply:
The plan is to adapterize this as well (I mentioned in the description, "support smth like `config.action_cable.async_executor = :fiber`"), so it will be possible to plug in any _executor_.

But I decided to leave it for follow-up PRs to focus only on refactoring what we already have right now. I'd better introduce _features_ one by one after we finalize the refactoring part.